lib/worldwide/address.rb

```diff
@@ -463,5 +475,32 @@ def concatenation_delimiter(concatenated_field, decorator, field_index)
         concatenated_field.blank? ? RESERVED_DELIMITER : "#{decorator}#{RESERVED_DELIMITER}"
       end
     end
+
+    def parse_fields(field_definitions, field_values)
+      field_definitions.each_with_index.with_object({}) do |(field, i), parsed_fields|
+        field_name = field["key"].underscore
+        field_value = if i < field_values.length
+          raw_field_value = field_values[i]
+          next_field = field_definitions[i + 1] unless last_field_in_arr?(field_definitions, i)
+          strip_decorators(next_field, raw_field_value)
+        else
+          ""
+        end
+        parsed_fields[field_name] = field_value
```

Comment:
Oh! Silly me, thanks Roch. I had to make a slight tweak so that we still return an empty string for the fields that are not found, but it's way simpler now


```
field_definitions.each_with_index.with_object({}) do |(field, i), parsed_fields|
    field_name = field["key"].underscore
    parsed_fields[field_name] = strip_decorators(field_definitions[i + 1], field_values[i]) || ""
end
```

Reply:
I like that it explicitly shows what fields exist for the country and what was not found, but I suppose a client should already be aware of which fields are relevant, and can infer what fields are blank based on their presence in our response. I'm fine with omitting the blank fields